packages/components/src/components/vertical-tab/vertical-tab-content-container.tsx

```diff
@@ -19,29 +40,29 @@ const SideNotes = ({ class_name, side_notes }) => {
     );
 };
 
-const ContentWrapper = ({ children, has_side_note }) => {
+const ContentWrapper = ({ children, has_side_note }: React.PropsWithChildren<TContentWrapper>) => {
     if (has_side_note) {
         return <div className='dc-vertical-tab__content-inner'>{children}</div>;
     }
-    return children;
+    return children as JSX.Element;
 };
 
-const Content = ({ is_routed, items, selected }) => {
+const Content = ({ is_routed, items, selected }: TContent) => {
     const selected_item = items.find(item => item.label === selected.label);
     const previous_selected_item = usePrevious(selected_item);
-    const TabContent = selected_item.value;
-    const [side_notes, setSideNotes] = React.useState(null);
+    const TabContent = selected_item?.value as React.ElementType;
+    const [side_notes, setSideNotes] = React.useState<React.ReactNode[] | null>(null);
 
-    const notes_array = [];
+    const notes_array: React.ReactNode[] = [];
 
-    const addToNotesQueue = React.useCallback(notes => {
+    const addToNotesQueue = React.useCallback((notes: React.ReactNode) => {
         notes_array.unshift(notes);
-        setSideNotes(notes);
+        setSideNotes(notes_array);
```

Comment:
@yauheni-deriv why did you change notes to notes_array here?

Reply:
I believe there were a typo, first we are adding item (notes) to array (notes_array) (line 59), and then setting it with useState. in this case this array (notes_array) is not used, and the value in useState (side_notes) is replaced to currents item (notes) instead of array.